doc/jinja-filters.md

```diff
@@ -0,0 +1,438 @@
+```
+
+##### Examples:
+```
+possible_items = ['item 1', 'item 2', 'item 3', 'item 4']
+list_items = [['item 1', 'item 3']]
+
+Input:                          Output:
+(possible_items, list_items)    "<ul><li>item 2</li><li>item 4</li></ul>"
+```
+
+
+## format_household_summary
+A merge of format_household_name and format_unordered_list in order to return a list of full names in "ul" and "li' tags 
+
+##### Parameters: 
+- names: A list containing lists of first, middle and last names [[all first names], [all middle names], [all last names]]
+
+##### Context:
+Used in Census to list everybody currently added to who lives here section. Used in a description.
+```
+<h2 class='neptune'>Your household includes:</h2> {{ [answers['first-name'], answers['middle-names'], answers['last-name']]|format_household_summary }}
+```
+
+##### Example:
+```
+Input: ([
+            ['fist1', 'first2', 'first3', 'first4'],
+            ['middle1', '', '', 'middle4'],
+            ['last1', 'last2', 'last3', '']
+        ])
+
+Ouput: "<ul>
+            <li>first1 middle1 last1</li>
+            <li>first2 last2</li>
+            <li>first3 last3</li>
+            <li>first4 middle4</li>
+       </ul>"
+```
+
+## format_repeating_summary
+Similar to format_household_summary but can handle lists from a series of sources.
+
+##### Parameters: 
+- items: List of lists, If there is a third level of lists, these will be zipped together e.g. [['John', 'Smith'], [['Jane', 'Sarah'], ['Smith', 'Smith']]]
+- delimiter: Default space.
+
+##### Context:
+Used in LMS to list everybody currently added to who lives here section. Used in a description.
+``` 
+{{ [[answers['primary-household-member-first-name'], answers['primary-household-member-last-name']], [answers['other-household-member-first-name'], answers['other-household-member-last-name']], [answers['student-household-member-first-name'], answers['student-household-member-last-name']]] | format_repeating_summary }}
+```
+
+##### Examples:
+```
+Input: ([
+            ['John', 'Smith'], 
+            [
+                ['Jane', 'Sarah'], 
+                ['Smith', 'Smythe']
+            ]
+        ])
+
+Ouput: "<ul>
+            <li>John Smith</li>
+            <li>Jane Smith</li>
+            <li>first3 last3</li>
+            <li>Sarah Smythe</li>
+       </ul>"
+
+       
+Input: ([
+            [
+                ['David', 'Sarah'], 
+                ['Smith', 'Smythe']
+            ]
+        ])
+
+Ouput: "<ul>
+            <li>David Smith</li>
+            <li>Sarah Smythe</li>
+       </ul>"
+
+
+Input: ([
+            ['', 
+             '51 Testing Gardens', 
+             '', 
+             'Bristol', 
+             'BS9 1AW'
+            ]
+        ], 
+        delimiter=', ')
+
+Ouput: "<ul>
+            <li>51 Testing Gardens, Bristol, BS9 1AW</li>
+       </ul>"
```

Comment:
Where do we use this filter for an address?

Reply:
We don't. I just wanted to show an example where the delimiter is changed and address seemed the best fit for that. I could remove it or have a much simpler example of Surname, First Name. Is this PR just about how it's currently used or how it could be used? Have I just opened myself up for even more changes?

Reply:
I'd say it's about how it's currently used, as we want it to drive the schema changes for piping.